app/shared/api-services/special-coverage/special-coverage-factory.js

```diff
@@ -101,6 +102,28 @@ angular.module('apiServices.specialCoverage.factory', [
             });
           },
           /**
+           * Load campaign data from Tunic endpoint
+           */
+          $loadTunicCampaign: function () {
+            if (_.isNumber(this.tunicCampaignId)) {
+              return $http.get(TunicConfig.buildBackendApiUrl('campaign/' + this.tunicCampaignId + '/')).then(function (result) {
+                return result.data;
```

Comment:
Per @kand the current interceptor will show a rudimentary error message, and it might not be ideal to add additional error handling logic at this specific location. Overall framework error handling approach needs some improvements.